Editorial review: Document CSS multicol column wrapping (#42934)
mdn/content: files/en-us/web/css/reference/properties/column-height/index.md

```css hidden live-sample___basic-column-height live-sample___column-playground
* {
  box-sizing: border-box;
}

html {
  font-family: Arial, Helvetica, sans-serif;
}

p {
  line-height: 1.5;
}

@supports not (column-height: 15em) {
  body::before {
    content: "Your browser does not support the 'column-height' property.";
    background-color: wheat;
    position: fixed;
    inset: 40% 0;
    height: fit-content;
    text-align: center;
    padding: 1rem 0;
  }
}
```

```css live-sample___basic-column-height
body {
  column-width: 200px;
  column-rule: 2px solid red;
  gap: 3em 2em;
  column-height: 95vh;

>   /* Default value if column-height is set; doesn't need to be set */

Reviewer: ```suggestion
```

#### HTML

Our markup is the same as the previous example, except that we provide a form containing two [`<input="range">`](/en-US/docs/Web/HTML/Reference/Elements/input/range) elements to allow you to change the `column-count` and `column-height` values. This functionality is powered by some basic JavaScript.

We won't show the markup, or the JavaScript, for brevity.

```html hidden live-sample___column-playground
<form>
  <div>
    <label for="columns">No. of columns: </label>
    <input type="range" min="3" max="6" value="3" id="columns" />
    <output id="columns-output">3</output>
  </div>
  <div>
    <label for="column-height">Column height: </label>
    <input type="range" min="10" max="30" value="20" id="column-height" />
    <output id="column-height-output">20em</output>
  </div>
</form>
```

#### CSS

The CSS is nearly identical to the previous example, except that we specify the number of columns with {{cssxref("column-count")}} rather than `column-width`, we don't set the `column-height` to the same height as the viewport, and we don't set the scroll snapping behavior.

> In the `body` rule, we set an initial `column-count` of `3` to divide our content into three columns, and set an initial `column-height` of `20em`. We also set the same `column-rule` and `gap` values as the previous example.

Reviewer: ```suggestion
We specify the {{cssxref("column-rule")}} and {{cssxref("gap")}} with the same values as the previous example. We don't specify a `column-width`; we will create a multi-column layout with the {{cssxref("column-count")}} property,  interactively setting the number of columns and the height of the column rows using JavaScript. We aren't including scroll snapping in this example.
```

```css live-sample___column-playground
body {
  column-count: 3;
  column-height: 20em;

  column-rule: 2px solid red;
  gap: 3em 2em;
}
```

```css hidden live-sample___column-playground
form {
  border: 1px solid black;
  background-color: white;
  padding: 10px;
  position: fixed;
  bottom: 1px;
  right: 1px;

> }

Reviewer: ```suggestion
  font-family: monospace;
}
```

Author: I did this with `<code>` elements in the HTML instead.

---
title: column-height
slug: Web/CSS/Reference/Properties/column-height
page-type: css-property
status:
  - experimental
browser-compat: css.properties.column-height
sidebar: cssref
---

{{SeeCompatTable}}

The **`column-height`** [CSS](/en-US/docs/Web/CSS) property specifies the height of the columns in a [CSS multi-column layout](/en-US/docs/Web/CSS/Guides/Multicol_layout).

The {{cssxref("columns")}} shorthand property can be used to set the `column-height`, {{cssxref("column-count")}}, and {{cssxref("column-width")}} property values in a single declaration.

## Syntax

```css

> /* <length> value */

Author: added

```css live-sample___column-playground
body {
  column-count: 3;
  column-height: 20em;
  column-rule: 2px solid red;
  gap: 3em 2em;
  padding: 0 2em;
}
```

```css hidden live-sample___column-playground
form {
  border: 1px solid black;
  background-color: white;
  padding: 10px;
  position: fixed;
  bottom: 1px;
  right: 1px;
}

form div {
  display: flex;
  gap: 1em;
  align-items: center;
  justify-content: space-between;
}
```

```js hidden live-sample___column-playground
const columnsRange = document.getElementById("columns");
const columnsOutput = document.getElementById("columns-output");
const columnHeightRange = document.getElementById("column-height");
const columnHeightOutput = document.getElementById("column-height-output");

columnsRange.addEventListener("input", () => {
  document.body.style.columnCount = columnsRange.value;
  columnsOutput.textContent = columnsRange.value;
});

columnHeightRange.addEventListener("input", () => {
  document.body.style.columnHeight = `${columnHeightRange.value}em`;
  columnHeightOutput.textContent = `${columnHeightRange.value}em`;
});
```

#### Result

The rendered result looks like this:

> 

Reviewer: ```suggestion

```
